webapp/channels/src/utils/browser_history.tsx

```diff
@@ -5,57 +5,27 @@ import {createBrowserHistory} from 'history';
 import type {History} from 'history';
 
 import {getModule} from 'module_registry';
+import {doBrowserHistoryPush, onBrowserHistoryPush} from 'utils/desktop_api';
 import {isServerVersionGreaterThanOrEqualTo} from 'utils/server_version';
 import {isDesktopApp, getDesktopVersion} from 'utils/user_agent';
 
 const b = createBrowserHistory({basename: window.basename});
 const isDesktop = isDesktopApp() && isServerVersionGreaterThanOrEqualTo(getDesktopVersion(), '5.0.0');
-
-type Data = {
-    type?: string;
-    message?: Record<string, string>;
-}
-
-type Params = {
-    origin?: string;
-    data?: Data;
-}
-
-window.addEventListener('message', ({origin, data: {type, message = {}} = {}}: Params = {}) => {
-    if (origin !== window.location.origin) {
-        return;
-    }
-
-    switch (type) {
-    case 'browser-history-push-return': {
-        if (message.pathName) {
-            const {pathName} = message;
-            b.push(pathName);
-        }
-        break;
-    }
-    }
-});
-
 const browserHistory = {
     ...b,
     push: (path: string | { pathname: string }, ...args: string[]) => {
         if (isDesktop) {
-            window.postMessage(
-                {
-                    type: 'browser-history-push',
-                    message: {
-                        path: typeof path === 'object' ? path.pathname : path,
-                    },
-                },
-                window.location.origin,
-            );
+            doBrowserHistoryPush(typeof path === 'object' ? path.pathname : path);
         } else {
             b.push(path, ...args);
         }
     },
 };
 
+if (isDesktop) {
+    onBrowserHistoryPush((pathName) => b.push(pathName));
```

Comment:
Actually I guess it's not necessary here because it's just a listener. I guess I wanted to save having a listener registered when it would be needed. Unfortunately running it means it'll default to the message passing case, so we'd be registering it unnecessarily.

1/5 on keeping it, but it's probably not a huge deal to just remove the check as well.

Reply:
I'm good with leaving that. I was just curious if you had a specific goal in mind with regards to having or not having those checks everywhere